Ghost role event code cleanup

coiax opened the pull request:
Most events that spawn ghost roles now inherit from /datum/round_event/ghost_role
- You can put ghosts in the "priority_candidates" list to make them get picked first (a `popleft` proc has been added for this purpose).
- Spawned mobs are put in the appropriately named `spawned_mobs` list for later tracking and handling.
- Spawning as a revenant, ninja, operative, **initial** xeno larva (the ones that spawn with the alien infestation event, this **DOES NOT** affect recruitment for xenolarvabursting) now uses the DO YOU WANT TO PLAY AS A GIANT SPIDER (as in, the tgui dialogue box, not the new style spider notification)
- So does spawning as a wizard 'diplomat'
- As a side effect because it made me cry, the "retry on insufficient ghosts" has been removed, because apparently it rarely actually worked

coiax committed:
[WIP] Ninja refactor and quality of life changes

- [ ] Fix ninja mask being hidden by hood
- [ ] Action buttons
- [ ] Ninjas can be spawned with custom objectives easily

coiax committed:
Adding the /datum/event/ghost_role subevent type

coiax committed:
I hate events retrying with insufficient pop

optimumtact commented on code/modules/events/abductor.dm:
```diff
@@ -9,55 +9,47 @@
+
+	if(!candidates.len < 2)
+		return FALSE
+	//Oh god why we can't have static functions
```
remove

duncathan replied:
this comment is legit because it explains why a new game mode datum has to be instantiated

optimumtact replied:
that was not clear, reword the comment so it is clear.

duncathan replied:
:+1:

coiax replied:
not my comment, was in the original

optimumtact commented on code/modules/ninja/ninja_event.dm:
```diff
@@ -7,7 +7,7 @@
 - The Ninja "Random" Event
 - Ninja creation code
-
+- coiax magic
```
remove

coiax replied:
that was my "make a commit to make a PR change"

duncathan commented on code/modules/events/abductor.dm:
```diff
@@ -9,55 +9,47 @@
+	if(!candidates.len < 2)
+		return FALSE
+	//Oh god why we can't have static functions
+	// I feel your pain, bro
```
this one, however, is not

duncathan commented on code/__HELPERS/lists.dm:
```diff
@@ -136,6 +136,11 @@
+/proc/popleft(list/L)
+	if(L.len)
+		. = L[1]
+		L.Remove(.)
```
`L.Cut(1,2)` is likely to be faster

coiax committed:
OH GOD WHY

coiax committed:
Let's just get rid of retries, it's silly

coiax committed:
Morph now uses the ghost_role event

coiax committed:
Operative does as well

Possibility of being bugged due to inability to run age checks on the
testing server, please be cautious.

coiax committed:
Ninjas, spawning, that sort of thing

Wizard diplomats now use the new ghost question system.

coiax committed:
Code review

coiax committed:
Merge branch 'master' into ninja

coiax committed:
Don't ask ghosts if we have enough selected

AnturK commented on code/modules/admin/verbs/one_click_antag.dm:
```diff
@@ -269,11 +242,11 @@
 
 /datum/admins/proc/makeAliens()
-	new /datum/round_event/alien_infestation{spawncount=3}()
+	new /datum/round_event/ghost_role/alien_infestation{spawncount=3}()
```
Could you make these an argument please, this notation really ticks me off outside of map definitions.

AnturK commented on code/modules/events/alien_infestation.dm:
```diff
@@ -1,73 +1,75 @@
+				vents += temp_vent
+
+	if(!vents.len)
+		// TODO do we throw an exception or just return a special
```
Trim the redundant comment please.

KorPhaeron commented:
@coiax 

Are you finishing this?

coiax committed:
Removes map notation from alien spawn

coiax committed:
Merge branch 'master' into ninja

Conflicts:
	tgstation.dme

coiax committed:
Removed comment
